content/sensu-core/0.29/guides/pre-compile-plugins.md

```diff
@@ -0,0 +1,148 @@
+---
+title: "Pre-Compiling Plugins"
+description: "Pre-Compiling Sensu Plugins for Distribution"
+product: "Sensu Core"
+version: "0.29"
+weight: 2
+next: ../intro-to-checks
+previous: ../overview
+menu:
+  sensu-core-0.29:
+    parent: guides
+---
+
+# Overview
+
+This guide will walk you through an example on how to pre-compile Sensu Plugins for distribution among clients.
+The goal of this guide is to showcase how you can quickly install plugins on ephemeral instances where time to compile can affect time to production.
+
+We'll be using `sensu-plugins-aws` as it requires `nokogiri`,  usually requiring a longer build time.
+
+# Prerequist
+
+Sources:
+* [GitHub - luislavena/gem-compiler: A RubyGems plugin that generates binary gems](https://github.com/luislavena/gem-compiler)
+
+This guide assumes the following:
+
+* Centos 7 platform
+* You have some way to store these artifacts and make them available for systems in your environment
+
+_NOTE: Compiled gems are tied to the Ruby version they are compiled for. If Sensu upgrade Ruby 2.4 -> Ruby 2.5, packages will need to be rebuilt._
+
+_NOTE: Compiled gems are tied to CPU architecture for which they are compiled on._
+
+## Installing The Build Toolchain
+
+1. Add the Sensu yum repository definition and install Sensu
+
+```bash
+printf '[sensu]
+name=sensu
+baseurl=https://repositories.sensuapp.org/yum/$releasever/$basearch/
+gpgcheck=0
+enabled=1' | sudo tee /etc/yum.repos.d/sensu.repo && yum install -y sensu
+```
+
+2. Install EPEL and Dev tools
+
+```bash
+sudo yum install -y epel-release
+sudo yum groupinstall -y "Development Tools"
+```
+3. Add Sensu's embedded bin to our path
+
+ ```bash
+ export PATH=$PATH:/opt/sensu/embedded/bin
+ ```
+
+## Installing a Gem to a tmp directory
+
+1. Install gem-compiler
+
+```bash
+gem install gem-compiler
+```
+
+2. Install the source gems into a cache directory
+
+```bash
+mkdir /tmp/gems
+gem install --no-ri --no-rdoc --install-dir /tmp/gems sensu-plugins-aws
+```
+
+Observing the `gem install` output, we can see that nokogiri and unf_ext build native extensions.
+These are the gems we will compile so we can install them from our own repository.
+
+### Compile the Gems for packaging
+
+1. We'll use `gem-compiler` to rebuild the gem packages with the included native extensions:
+
+```bash
+cd /tmp/gems/cache
+gem compile unf_ext-0.0.7.4.gem
```

Comment:
Get an error here:

```gem compile unf_ext-0.0.7.4.gem
Unpacking gem: 'unf_ext-0.0.7.4' in temporary directory...
ERROR:  While executing gem ... (Gem::InstallError)
    invalid gem: No such file or directory @ rb_sysopen - unf_ext-0.0.7.4.gem```

@Blue0ctober we probably should do a pairing to speed this up. Let me know!